core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -513,60 +499,57 @@ private void registerBuiltinFunctionScoreParsers() {
     }
 
     private void registerBuiltinQueryParsers() {
-        registerQuery(MatchQueryBuilder.PROTOTYPE::readFrom, MatchQueryBuilder::fromXContent, MatchQueryBuilder.NAME,
-                "match_phrase", "matchPhrase", "match_phrase_prefix", "matchPhrasePrefix", "matchFuzzy", "match_fuzzy", "fuzzy_match");
-        registerQueryParser(MultiMatchQueryParser::new);
-        registerQueryParser(NestedQueryParser::new);
-        registerQueryParser(HasChildQueryParser::new);
-        registerQueryParser(HasParentQueryParser::new);
-        registerQueryParser(DisMaxQueryParser::new);
-        registerQueryParser(IdsQueryParser::new);
-        registerQueryParser(MatchAllQueryParser::new);
-        registerQueryParser(QueryStringQueryParser::new);
-        registerQueryParser(BoostingQueryParser::new);
+        registerQuery(MatchQueryBuilder.NAMES, MatchQueryBuilder.PROTOTYPE::readFrom, MatchQueryBuilder::fromXContent);
```

Comment:
The idea is to replace calling readFrom against the prototype with calling a new constructor for each query that takes a StreamInput as argument. Depending on where those common fields are read/written they may have to be moved, which is the part that breaks serialization as a constructor can only call super from its first line.